navigation/index.tsx

```diff
@@ -117,187 +113,189 @@ if (Platform.OS === "android") {
                   >
-                    <AntDesign name="left" size={24} color="black" />
-                  </TouchableOpacity>
-                  <Text style={fonts.pageHeading3}>New Listing</Text>
-                  <TouchableOpacity
-                    activeOpacity={pressedOpacity}
+                    <TouchableOpacity
+                      activeOpacity={pressedOpacity}
+                      style={{
+                        position: "absolute",
+                        right: 20,
+                        alignItems: "center",
+                        justifyContent: "center",
+                        height: 50,
+                        width: 50,
+                      }}
+                      onPress={() => navigation.goBack()}
+                    >
+                      <AntDesign
+                        name="close"
+                        size={24}
+                        color="black"
+                        style={{ marginStart: 18 }}
+                      />
+                    </TouchableOpacity>
+                    <Text style={fonts.pageHeading3}>New Listing</Text>
+                  </View>
+                );
+              },
+            })}
+            component={NewPostImage}
+          />
+          <Stack.Screen
+            name="NewPostDetail"
+            options={({ navigation }) => ({
+              headerShadowVisible: false,
+              headerStyle: styles.headerNoShadow,
+              headerBackVisible: false,
+              headerTitleAlign: "center",
+              header: () => {
+                return (
+                  <View
                     style={{
-                      position: "absolute",
-                      right: 20,
-                      alignItems: "center",
+                      height: 70,
+                      backgroundColor: "#ffffff",
                       justifyContent: "center",
-                      width: 50,
-                      height: 50,
+                      alignItems: "center",
+                      marginTop: Platform.OS === "ios" ? 35 : 0,
                     }}
-                    onPress={() => navigation.navigate("Root")}
                   >
-                    <AntDesign name="close" size={24} color="black" />
-                  </TouchableOpacity>
-                </View>
-              );
-            },
-          })}
-          component={NewPostDetail}
-        />
+                    <TouchableOpacity
+                      activeOpacity={pressedOpacity}
+                      style={{
+                        position: "absolute",
+                        left: 24,
+                        alignItems: "center",
+                        justifyContent: "center",
+                        width: 50,
+                        height: 50,
+                      }}
+                      onPress={() => navigation.goBack()}
+                    >
+                      <AntDesign name="left" size={24} color="black" />
```

Comment:
Since this icon and and the one below are the same type of icon and the same size, perhaps we could also pull this out into a `NewPostIcon` component but that's up to you